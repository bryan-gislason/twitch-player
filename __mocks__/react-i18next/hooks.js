const { get } = require("lodash");

const i18n = {
  language: "en-US",
  languages: ["en-US", "en"],
  dir: () => "ltr",
  on: jest.fn(),
  off: jest.fn(),
  changeLanguage: jest.fn(),
};

const englishResources = require("@twitch-player/i18n/locales/en/translation.json");

module.exports = {
  getI18n: () => i18n,
  useTranslation: () => [i18nKey => get(englishResources, i18nKey), i18n],
};
