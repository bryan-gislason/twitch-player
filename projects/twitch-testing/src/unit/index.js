import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import { OverridesProvider } from "@twitch-player/ui/dist/context";
import React from "react";
import { act } from "react-dom/test-utils";
import { MemoryRouter as Router } from "react-router";
import { fireEvent, render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>
      <OverridesProvider>
        <Router>{node}</Router>
      </OverridesProvider>
    </ThemeProvider>,
    ...options
  );
};

fireEvent["press"] = (node, init) => {
  fireEvent.mouseDown(node, init);
  fireEvent.mouseUp(node, init);
};

fireEvent["resize"] = (node, init) => {
  const { target: { left = 0, top = 0, width = 0, height = 0 } = {} } = init;

  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const eventInit = { bubbles: false, cancelable: false };
  const event = new Event(node, eventInit);

  act(() => {
    fireEvent(node, event);
    // eslint-disable-next-line no-undef
    jest.runAllTimers();
  });
};

export * from "react-testing-library";

export { customRender as render };
