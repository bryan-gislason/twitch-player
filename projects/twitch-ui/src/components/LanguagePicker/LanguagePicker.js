import { supportedLanguages, useTranslation } from "@twitch-player/i18n";
import PropTypes from "prop-types";
import React, { memo, useCallback } from "react";
import Picker from "../Picker";

const LanguagePicker = memo(({ testID = "language-picker" }) => {
  const [, i18n] = useTranslation();

  const lng = i18n.languages.find(language =>
    supportedLanguages.hasOwnProperty(language)
  );

  const changeLanguage = useCallback(language => {
    i18n.changeLanguage(language);
  }, []);

  return (
    <Picker
      values={supportedLanguages}
      selectedValue={lng}
      testID={testID}
      onValueChange={changeLanguage}
    />
  );
});

LanguagePicker.propTypes = {
  testID: PropTypes.string,
};

export default LanguagePicker;
