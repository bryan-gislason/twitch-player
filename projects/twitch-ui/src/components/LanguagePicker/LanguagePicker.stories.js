import { storiesOf } from "@storybook/react";
import React from "react";
import LanguagePicker from ".";

storiesOf("Components|LanguagePicker", module).add("default", () => (
  <LanguagePicker />
));
