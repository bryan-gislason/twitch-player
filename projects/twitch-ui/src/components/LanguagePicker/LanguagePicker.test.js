import { fireEvent, render } from "@twitch-player/testing/dist/unit";
import React from "react";
import { getI18n } from "react-i18next";
import LanguagePicker from "./LanguagePicker";

describe("LanguagePicker", () => {
  let instance;

  beforeEach(() => {
    instance = render(<LanguagePicker />);
  });

  it("renders a picker with all supported languages", () => {
    expect(instance.getByTestId("language-picker")).toMatchSnapshot();
  });

  describe("when changed", () => {
    beforeEach(() => {
      fireEvent.change(instance.getByTestId("language-picker"), {
        target: { value: "es" },
      });
    });

    it("changes the current language", () => {
      expect(getI18n().changeLanguage).toHaveBeenCalledWith("es");
    });
  });
});
