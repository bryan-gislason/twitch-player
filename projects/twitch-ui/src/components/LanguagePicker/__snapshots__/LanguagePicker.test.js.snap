// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguagePicker renders a picker with all supported languages 1`] = `
<select
  class="rn-fontFamily-14xgk7a rn-fontSize-7cikom rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-width-13qz1uu"
  data-focusable="true"
  data-testid="language-picker"
  style="background-color: rgb(250, 249, 250); color: rgb(25, 23, 28);"
>
  <option
    value="en"
  >
    English
  </option>
  <option
    value="es"
  >
    Español
  </option>
  <option
    value="fr"
  >
    Français
  </option>
  <option
    value="de"
  >
    Deutsch
  </option>
  <option
    value="it"
  >
    Italiano
  </option>
  <option
    value="he"
  >
    עִבְרִית
  </option>
  <option
    value="ar"
  >
    العَرَبِيَّة‎
  </option>
</select>
`;
